### PR TITLE
STRWEB-23 Postcss-* upgrade

### PR DESCRIPTION
**Pull request by JohnC-80**
Upgrade `postcss` in support of upgrade `@storybook/react`.

Refs [STRWEB-23](https://issues.folio.org/browse/STRWEB-23)

### CODE REVIEW

**Review by mkuklis — APPROVED** [review]

**Review by zburke — APPROVED** [review]
Yay! Thank you.

**Review by zburke — CHANGES_REQUESTED** [review]
Alas, tests pass, but when running this branch in a workspace, the build fails:
```
$ yarn local
yarn run v1.22.10
$ f=stripes.config.js; test -f $f.local && f=$f.local; echo Using config $f; yarn stripes serve $f
Using config stripes.config.js.local
$ /Users/zburke/temp/strweb-23/node_modules/.bin/stripes serve stripes.config.js.local
Waiting for webpack to build...
Listening at http://localhost:3000
node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

[Error: ENOENT: no such file or directory, open '/Users/zburke/temp/strweb-23/platform-core/lib/variables.css'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/Users/zburke/temp/strweb-23/platform-core/lib/variables.css'
}
error Command failed with exit code 1.
```
Maybe the scoping rules have changed and variables.css is being looked for in the wrong place? That file exists in stripes-components, but it's being searched for where the build is generated, in the platform repository.

File: webpack.config.cli.dev.js [hunk 1]
```diff
@@ -53,28 +53,31 @@ devConfig.module.rules.push({
...
-          postCssMediaMinMax(),
-          postCssColorFunction(),
-        ],
+        postcssOptions: {
+          plugins: [
+            postCssImport(),
+            autoprefixer(),
+            postCssCustomProperties({
+              preserve: false,
+              importFrom: './lib/variables.css'
```
Comment (zburke) [inline webpack.config.cli.dev.js:72]: On #master, `lib/variables.css` was replaced with `[locateCssVariables()]` in #4. Did you undo that intentionally?

**Comment by zburke** [comment]
Restoring `locateCssVariables` allows the build to succeed, sort of: you get a bundle (yay!), but it's beset with some fatal CSS errors. I fixed the [straight up typo in ui-users](folio-org/ui-users/pull/1830) but I don't know how to handle the other two (one each in stripes-core and stripes-smart-components): 
```
ERROR in ../node_modules/@folio/stripes-core/src/components/MainNav/NavButton/NavButton.css 
(../node_modules/css-loader/dist/cjs.js??
ref--11-1!../node_modules/postcss-loader/dist/cjs.js??
ref--11-2!../node_modules/@folio/stripes-core/src/components/MainNav/NavButton/NavButton.css)
Module not found: Error: Can't resolve './@folio/stripes-components/lib/sharedStyles/interactionStyles.css' in 
'/Users/zburke/temp/strweb-23/node_modules/@folio/stripes-core/src/components/MainNav/NavButton'

ERROR in ../node_modules/@folio/stripes-smart-components/lib/SearchAndSort/components/SearchButton/SearchButton.css 
(../node_modules/css-loader/dist/cjs.js??
ref--11-1!../node_modules/postcss-loader/dist/cjs.js??
ref--11-2!../node_modules/@folio/stripes-smart-components/lib/SearchAndSort/components/SearchButton/SearchButton.css)
Module not found: Error: Can't resolve './@folio/stripes-components/lib/sharedStyles/interactionStyles.css' in 
'/Users/zburke/temp/strweb-23/node_modules/@folio/stripes-smart-components/lib/SearchAndSort/components/SearchButton'
```
Those are both in `composes: some-value from "@folio/stripes-components/lib/some/path.css"` statements. IOW, this new version of postcss seems not to like `composes ... from "@package"` statements. `composes ... from "./file"` statements seem to be OK, but I didn't have quite enough to go on to search the manual and figure out why package-based imports don't work. Any ideas?

File: webpack.config.cli.dev.js (hunk 1 above)
Comment (JohnC-80) [reply webpack.config.cli.dev.js:72]: I definitely did not intend this... committing the correction...

**Comment by zburke** [comment]
Sweet; those recent commits fix the build issues, but now stripes-core isn't finding its icons so something still isn't quite right.
Screen Shot 2021-08-30 at 11 33 19 AM (2)

**Review by zburke — APPROVED** [review]
LGTM now! In concert with the [`storybook-6` branch on `stripes-components`](https://github.com/folio-org/stripes-components/pull/1598), all app icons appear correctly.

**Pull request merged**